Find a functionally equivalent code:android.content.res.TypedArray.getColorStateList:COMMENT
Method Modifier: public      
Comment:/**
 * Retrieve the ColorStateList for the attribute at <var>index</var>.
 * The value may be either a single solid color or a reference to
 * a color or complex {@link android.content.res.ColorStateList} description.
 *
 * @param index Index of attribute to retrieve.
 *
 * @return ColorStateList for the attribute, or null if not defined.
 */

@@ -1,6 +1,12 @@
 {
+    if (mRecycled) {
+        throw new RuntimeException("Cannot make calls to a recycled instance!");
+    }
     final TypedValue value = mValue;
     if (getValueAt(index * AssetManager.STYLE_NUM_ENTRIES, value)) {
+        if (value.type == TypedValue.TYPE_ATTRIBUTE) {
+            throw new RuntimeException("Failed to resolve attribute at index " + index);
+        }
         return mResources.loadColorStateList(value, value.resourceId);
     }
     return null;

